From 970966b03e947332197a2982d59fc13142ab5b65 Mon Sep 17 00:00:00 2001
From: Peter Swain <pswain@marvell.com>
Date: Thu, 24 Jan 2019 01:24:43 -0800
Subject: [PATCH 062/138] mmc: cavium: skip unavailable slots

commit f39a7356f85b93a9a58db03c00aa090230dfc7cf from
git@git.assembla.com:cavium/WindRiver.linux.git

Skip devtree mmc-slot nodes marked unavailable
(those with a status property that is not "ok")

Note that a missing 'status' property marks the node
available, nothing need be added to activate it.

This allows multiple alternate entries in firmware
devtree, with dormant nodes marked (eg) status="off",
to be boot-time enabled without rewriting the whole fdt,
by simple u-boot commands like
  fdt rm $mmc/mmc-slot@2 status
enabling an mmc2 entry that is normally disabled

Change-Id: Ib680a2bc1dd58bebaabcbb47f2560c00f32ab250
Signed-off-by: Peter Swain <pswain@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/8544
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/mmc/host/cavium-octeon.c   | 2 +-
 drivers/mmc/host/cavium-thunderx.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/host/cavium-octeon.c b/drivers/mmc/host/cavium-octeon.c
index 22aded1065ae..d7d70798cb7c 100644
--- a/drivers/mmc/host/cavium-octeon.c
+++ b/drivers/mmc/host/cavium-octeon.c
@@ -277,7 +277,7 @@ static int octeon_mmc_probe(struct platform_device *pdev)
 	platform_set_drvdata(pdev, host);
 
 	i = 0;
-	for_each_child_of_node(node, cn) {
+	for_each_available_child_of_node(node, cn) {
 		host->slot_pdev[i] =
 			of_platform_device_create(cn, NULL, &pdev->dev);
 		if (!host->slot_pdev[i]) {
diff --git a/drivers/mmc/host/cavium-thunderx.c b/drivers/mmc/host/cavium-thunderx.c
index 261d63639466..2b38d4ac4074 100644
--- a/drivers/mmc/host/cavium-thunderx.c
+++ b/drivers/mmc/host/cavium-thunderx.c
@@ -185,7 +185,7 @@ static int thunder_mmc_probe(struct pci_dev *pdev,
 	 */
 	thunder_calibrate_mmc(host);
 
-	for_each_child_of_node(node, child_node) {
+	for_each_available_child_of_node(node, child_node) {
 		/*
 		 * mmc_of_parse and devm* require one device per slot.
 		 * Create a dummy device per slot and set the node pointer to
-- 
2.17.1

